protocols/imap/src/main/java/org/apache/james/imap/processor/base/SelectedMailboxImpl.java

```diff
@@ -396,6 +392,28 @@ private void mailboxEvent(MailboxEvent mailboxEvent) {
         }
     }
 
+    private void updateApplicableFlags(FlagsUpdated messageEvent) {
```

Comment:
given the complexity of this method I would have written unit tests on it to have a good coverage.

Reply:
Well, that is orthogonal to the PR goal, I just relied on IDE refactoring tools here (that's why I did a separated commit BTW)

There is no code change within that method.

Reply:
> Well, that is orthogonal to the PR goal, I just relied on IDE refactoring tools here (that's why I did a separated commit BTW)
> 
> There is no code change within that method.

It's hard to see what changed because of the extraction but I did the exercise of doing an actual diff:

![Screenshot from 2020-06-09 11-03-00](https://user-images.githubusercontent.com/708318/84128426-1a9efa80-aa30-11ea-950a-1d51ea596f19.png)

There's definitely some changes to the existing code that is probably tested only for some cases with higher level tests.

So I don't agree with that practice: every time somebody touch a method that is not trivial, the risk of introducing bugs to it is likely, It's a very common rule to say that only code with good test coverage should be refactored.

Reply:
https://github.com/linagora/james-project/pull/3367/commits/027fc63d40f71d5ba0c1f61b597ecf99f04f90b4 entitled `JAMES-3177 SelectedMailboxImpl: extract applicable flags update ` do not match the diff you presented me.

Well, sure, we can postpone any improvement until we have a perfect changeset, and test coverage, however it delays adoption of some reasonable fixes/refactoring IMO.

Reply:
Who decide what is "reasonable" or not?

Being against some changes in an untested class doesn't mean "postpone any improvement"

I don't understand why you argue against writing some unit tests (it's very fast to do).

If you want to minimize the number of test cases, you can even split that function into smaller functions so you don't have to test what you didn't change.

Would you accept a change from any random contributor who doesn't care to demonstrate (s)he didn't break existing behavior?